lib/encoding/json.toit

```diff
@@ -100,53 +133,70 @@ decode-stream reader:
   d := StreamingDecoder
   return d.decode-stream reader
 
-
+/**
+Deprecated.  Use the top level json.encode or json.stringify functions
+  instead.
```

Comment:
Given that we pass the encoder to the converter, we might not want to deprecate it yet.